src/AutoRest.CSharp/Common/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -328,11 +328,10 @@ private static void DeserializeIntoObjectProperties(this CodeWriter writer, IEnu
                              true) //https://github.com/Azure/autorest.csharp/issues/922
                     {
                         var emptyStringCheck = GetEmptyStringCheckClause(property, itemVariable, shouldTreatEmptyStringAsNull);
-                        if (Configuration.AzureArm && property.ValueType?.Equals(typeof(Uri)) == true)
+                        if (property.PropertySerializations is null)
```

Comment:
nit: any reason we don't use `!property.IsClientFlattenedProperty`?

Reply:
In the meantime I moved the comment on `IsClientFlattenedProperty` to `PropertySerializations` so that one could know what this property is for

Reply:
This is per Alexander's comment. `PropertySerialization is null` might be better here.